tests/test_multieurlex_utils.py

```diff
@@ -53,46 +63,223 @@ def test_extract_articles_multi_lang():
+    dummy_im1 = PILImage.fromarray(
+        np.random.randint(0, 255, (10, 10, 3)).astype(np.uint8)
+    )
+    dummy_im2 = PILImage.fromarray(
+        np.random.randint(0, 255, (10, 10, 3)).astype(np.uint8)
+    )
+
+    with patch("arc_spice.data.multieurlex_utils.GeneratorFromStrings") as mock_gen:
+        mock_gen.return_value = [(dummy_im1, "target1"), (dummy_im2, "target2")]
+        output = multieurlex_utils.make_ocr_data(row)
+
+    assert output == {
+        "ocr_images": (dummy_im1, dummy_im2),
+        "ocr_targets": ("target1", "target2"),
+    }
+
+
+def _check_keys_untouched(
+    original_dataset: datasets.Dataset,
+    dataset: datasets.Dataset,
+    indices_kept: list[int],
+    ignore_keys=list[str],
+) -> None:
+    # check remaining keys are untouched
+    for key in dataset.features:
+        if key not in ignore_keys:
+            assert dataset[key] == [original_dataset[key][i] for i in indices_kept]
+
+
 def test_load_multieurlex_en():
     data_dir = f"{TEST_ROOT}/testdata/multieurlex_test_en"
     level = 1
     languages = ["en"]
     drop_empty = True
 
     ds = datasets.load_from_disk(data_dir)
-    with patch("arc_spice.data.multieurlex_utils.load_dataset", return_value=ds):
+    expected_keys = {"celex_id", "text", "labels"}
+    expected_non_empty_indices = [0, 1, 3, 4]
+    text_expected_non_empty_indices = [i + 1 for i in expected_non_empty_indices]
+    with patch(
+        "arc_spice.data.multieurlex_utils.datasets.load_dataset", return_value=ds
+    ):
         dataset_dict, metadata = multieurlex_utils.load_multieurlex(
             data_dir=data_dir, level=level, languages=languages, drop_empty=drop_empty
         )
         assert len(dataset_dict) == 3
-        assert len(dataset_dict["train"]) == 4  # 5 items, 1 is empty so dropped
-        assert len(dataset_dict["validation"]) == 4  # 5 items, 1 is empty so dropped
-        assert len(dataset_dict["test"]) == 4  # 5 items, 1 is empty so dropped
-        assert dataset_dict["train"]["text"] == [
-            f"{multieurlex_utils.ARTICLE_1_MARKERS['en']} Some text after the marker {i}"  # noqa: E501
-            for i in [1, 2, 4, 5]  # 3 dropped
-        ]
+        for split in ["train", "validation", "test"]:
+            assert set(dataset_dict[split].features.keys()) == expected_keys
+            assert len(dataset_dict[split]) == 4  # 5 items, 1 is empty so dropped
+            assert dataset_dict[split]["text"] == [
+                f"{multieurlex_utils.ARTICLE_1_MARKERS['en']} Some text after the marker {i}"  # noqa: E501
+                for i in text_expected_non_empty_indices  # 3 dropped
+            ]
+            _check_keys_untouched(
+                original_dataset=ds[split],
+                dataset=dataset_dict[split],
+                indices_kept=expected_non_empty_indices,
+                ignore_keys=["text"],
+            )
 
 
-def test_load_multieurlex_for_translation():
+def test_load_multieurlex_multi_lang():
```

Comment:
I realised that the previous tests didn't actually cover everything. It now checks that other keys exist. There are alsso separate tests now for the `load_multieurlex_for_pipeline` method. Including one that checks the OCR output